### PR TITLE
[DOCS] Reformats cluster pending tasks API

### PR DESCRIPTION
**Pull request by szabosteve**
Relates to elastic/docs#937 and https://github.com/elastic/elasticsearch/issues/45197.

This PR updates the cluster pending tasks API to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Resources:
* [Cluster pending stats API spec](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/cluster.pending_tasks.json)
* [Cluster pending stats API doc](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/cluster-pending.html)

### CODE REVIEW

**Comment by szabosteve** [comment]
@jrodewig I commented out one of the response body parameters as I didn't have enough input to come up with a fair-enough description. Could you please provide some input or point to somebody who can help with it?

File: docs/reference/cluster/pending.asciidoc [hunk 1]
```diff
@@ -1,24 +1,65 @@
...
-The pending cluster tasks API returns a list of any cluster-level changes
-(e.g. create index, update mapping, allocate or fail shard) which have not yet
-been executed.
+Returns cluster-level changes that have not yet been executed.
+
+
+[[cluster-pending-api-request]]
+==== {api-request-title}
+
+GET /_cluster/pending_tasks
```
Comment (jrodewig) [inline docs/reference/cluster/pending.asciidoc]:
```suggestion
`GET /_cluster/pending_tasks`
```

File: docs/reference/cluster/pending.asciidoc [hunk 2]
```diff
@@ -1,24 +1,65 @@
...
+    (object) A list of pending tasks.
+    
+`insert_order`::
+    (integer) The number that represents when the task has been inserted into 
+    the task queue.
+    
+`priority`::
+    (string) The priority of the pending task.
+    
+// `source`::
```
Comment (jrodewig) [inline docs/reference/cluster/pending.asciidoc]: This looks like a description of the ongoing task.

You can confirm this definition by reaching out to the @elastic/es-core-features team here or on Slack.

**Comment by jrodewig** [comment]
@szabosteve  Looks good overall. Left a few minor comments.

Generally, I think we can leave off the response body sections going forward.

I can follow up and add those as part of a separate project.

File: docs/reference/cluster/pending.asciidoc (hunk 2 above)
Comment (szabosteve) [reply docs/reference/cluster/pending.asciidoc]: Thanks! I double-checked with them and amended the text accordingly.

**Comment by szabosteve** [comment]
Backported to 7.x and 7.3.

**Pull request merged**